### ISSUE TITLE
Insert into Identity colums 

### ISSUE THREAD

**Issue by shishmak** [linq2db/linq2db#1938]
Hello.

My SQL qwery don't contains Id value on insert when property Id has [Identity] atribute.
But how can I insert into db service-side genetated guid?

model:
` public partial class ExDbModel : LinqToDB.Data.DataConnection
 {
    [Table(Schema = "billing", Name = "TestTable")]
    public partial class billing_TestEntity : BEntity
    {
        [Column(), NotNull] public string Name { get; set; } // text
    }
    public class BEntity
    {
        [Column("id"), PrimaryKey, Identity] public Guid Id { get; set; }
    }
 }`

insert:
`using (var db = new ExDbModel())
                try
                {
                    var z = new billing_TestEntity { Id = Guid.NewGuid(), Name = "T1" };
                    db.InsertAsync(z).Wait();
                }
                catch (Exception e)
                {
                    Console.WriteLine(e.Message);
                }
`

I found this way (but I think there should be a simpler solution)
`
private static BulkCopyOptions _bulkCopyOptions = new BulkCopyOptions { KeepIdentity = true, BulkCopyType = BulkCopyType.MultipleRows, MaxBatchSize = 10 };
...
db.BulkCopy(_bulkCopyOptions, new billing_TestEntity[] { new billing_TestEntity { Id = Guid.NewGuid(), Name = "T1" } });
`
Сan you add similar options to insert methods? or I must use Bulk insert anywhere? (I cann't delete [identity] in prop and I need server-side generation guid Id in this method group)

###Test Environment details
linq2db version: 2.9.1
Database Server: PostgreSQL 9.6
Database Provider: Npgsql.EntityFrameworkCore.PostgreSQL 2.2.4
Operating system: Windows 10
.NET Framework: Net Core 2.2

**Comment by MaceWindu** [comment]
To insert server-side value to field, marked with `Identity` attribute, you should use Insert/Set methods with explicit field assignment: https://stackoverflow.com/a/53335783/1641529

**Comment by sdanyliv** [comment]
Try these variants:
```cs
using (var db = new ExDbModel())
{
     db.GetTable<billing_TestEntity>()
         .Insert(() => new billing_TestEntity { Id = Guid.NewGuid(), Name = "T1" });

     db.GetTable<billing_TestEntity>()
         .Value(e => e.Id, Guid.NewGuid())
         .Value(e => e.Name, "T1")
         .Insert();
}
```

**Comment by MaceWindu** [comment]
Wait a minute... we don't support non-numeric identities right now, so putting Identity attribute to Guid field doesn't make sense so just remove it.

**Comment by shishmak** [comment]
> Wait a minute... we don't support non-numeric identities right now, so putting Identity attribute to Guid field doesn't make sense so just remove it.

in my project I use both "EF core" and "linq2db.EntityFrameworkCore" and [Identity] came from EF model.

**Comment by sdanyliv** [comment]
@shishmak, why you do not save entity via EF?

**Comment by shishmak** [comment]
> @shishmak, why you do not save entity via EF?
mmm, I think it's not good idea mix 2 ORMs in one service

**Comment by sdanyliv** [comment]
> in my project I use both "EF core" and "linq2db.EntityFrameworkCore" 
:)

How you define model in EF and how do you reuse EF Model in linq2db?
If you set `ValueGenerated.OnAdd` on column, we mark this column as Identity, but maybe it is wrong?

**Comment by shishmak** [comment]
> How you define model in EF and how do you reuse EF Model in linq2db?
> If you set `ValueGenerated.OnAdd` on column, we mark this column as Identity, but maybe it is wrong?

@sdanyliv wait a little time while I collect the data to answer your questions

**Comment by iColdoCatalin** [comment]
I have the same issue: trying to insert with identity. I've tryied everything as 
`db.GetTable<Test>()
         .Value(e => e.Id, 5)
         .Value(e => e.Name, "T1")
         .Insert();`
but they all fail with same error "Cannot insert identity ...."
has anyone found a valid solution?
In our scenario we need to move data from a table to another and need to keep the same PK

**Comment by sdanyliv** [comment]
Yes I know. Which database do you use and how do you plan to copy tables? Within the same database? Or between two different servers?

**Comment by iColdoCatalin** [comment]
@sdanyliv  I need to copy same table between 2 different dbs on different servers. (SQL Server)
On the second server the tables have less collumns but need to have same Ids since there are tables referencing it.
Any solution? I'm currently testing the bulk copy as mentioned by @shishmak  and seems to work.

**Comment by sdanyliv** [comment]
So, then only `BulkCopy`. Pas as parameter  `new BulkCopyOptions { KeepIdentity = true }`
